content/en/blog/_posts/2022-10-05-current-state-2019-third-party-audit.md

```diff
@@ -0,0 +1,117 @@
+2019](https://github.com/kubernetes/sig-security/tree/main/sig-security-external-audit/security-audit-2019).
+
+## Motivation
+
+[Craig Ingram](https://github.com/cji) has graciously attempted over the years to keep track of the 
+status of the findings reported in the last audit in this issue:
+[kubernetes/kubernetes#81146](https://github.com/kubernetes/kubernetes/issues/81146)
+. This blog post will attempt to dive deeper into this, address any gaps
+in tracking and become a point in time summary of the state of the
+findings reported from 2019.
+
+This blog should also help folks gain confidence through transparent
+communication of work done by the community to address these findings and bubble up
+any findings that need help from community contributors.
+
+## Current State
+
+The status of each issue / finding here is represented in a best effort manner.
+Authors do not claim to be 100% accurate on the status and welcome any
+corrections or feedback if the current state is not reflected accurately by
+commenting directly on the relevant issue.
+
+| **\#** | **Title**                                                                   | **Issue**                                                       | **Status**                                                                                                                                                                                                                      |
+|--------|-----------------------------------------------------------------------------|-----------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 1      | hostPath PersistentVolumes enable PodSecurityPolicy bypass                  | [#81110](https://github.com/kubernetes/kubernetes/issues/81110) | closed, addressed by [kubernetes/website#15756](https://github.com/kubernetes/website/pull/15756)                                                                                                                               |
+| 2      | Kubernetes does not facilitate certificate revocation                       | [#81111](https://github.com/kubernetes/kubernetes/issues/81111) | duplicate of [#18982](https://github.com/kubernetes/kubernetes/issues/18982) and needs a KEP                                                                                                                                    |
+| 3      | HTTPS connections are not authenticated                                     | [#81112](https://github.com/kubernetes/kubernetes/issues/81112) | Largely left as an end user exercise in setting up the right configuration                                                                                                                                                       |
```

Comment:
I don't see f/u for this below?

If you peel the layers I disagree that the root issue is only [an end user exercise](https://github.com/kubernetes/kubernetes/issues/81112#issuecomment-520162629)

From this particular discussion @liggitt identified that even with all the flags configured as documented, ["the apiserver does not verify the TLS serving certificate presented by the backend"](https://github.com/kubernetes/kubernetes/issues/81112#issuecomment-520490993) and further proposes (maybe there is a KEP for this already?) the ["pod spec hostname/subdomain API fields"](https://github.com/kubernetes/kubernetes/issues/81112#issuecomment-520491760) for authentication.

But TLS doesn't (without extensions proposed) generally verify hostname (or IP) for the authentication, eg. ["TLS does not protect the source or destination IP and port address of the TCP connection. As is stated in the TLS RFC, TLS is application layer independent."](https://alpaca-attack.com/#question-answer) - just the crypto of the certificates which includes the signature of the x509 CN attribute _which can be anything_ so there has to be agreement on the app level protocol to use, which is what is recommended above. So just the use of [client cert mutual auth TLS](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#x509-client-certs) doesn't in and of itself solve all problems.  (But it probably solves the problem originally identified.)

To me the tangential but probably most important issue here is [so many proxies!](https://kubernetes.io/docs/tasks/access-application-cluster/access-cluster/#so-many-proxies) and maybe a lack of documentation of the design forces and use cases of each sufficient to enable the operator to clearly articulate when to use each and how. (or maybe just lack of time to find all the documentation including this author ;) And maybe more generally [understanding of the issues re: authn vs authz](https://kubernetes.io/docs/reference/access-authn-authz/kubelet-authn-authz/).

Reply:
To me, @PushkarJ is correct. The _actual situation_ is that the project is treating this as an issue for end users.

@sunstonesecure-robert you might not like the situation. However, this is not really the best place for that feedback. This blog article is perhaps like a newpaper article about what banks have done about security. Even if an editor of that newspaper is a customer at the same bank, I wouldn't expect the editor to give feedback to the journalist about their reporting.

Feel free to suggest new wording @sunstonesecure-robert but remember that your new wording should aim to better capture the existing situation. If, separately, you want to add comments to any of the issues that are open, you're welcome to. Help with those issues is also very welcome.

Reply:
Thanks @sftim - agree with your analysis and ask for moving the conversation outside of this blog PR. 

@sunstonesecure-robert thanks for the detailed feedback. I have added a link to your comment to the original issue so if needed folks can continue the discussion with you there: https://github.com/kubernetes/kubernetes/issues/81112#issuecomment-1266123234

With a clear path for future follow up I am gonna mark this as resolved.